test/metabase/test/util.clj

```diff
@@ -602,8 +606,27 @@
            .getZone
            .getID)))))
 
+(defmulti with-model-cleanup-additional-conditions
+  "Additional conditions that should be used to restrict which instances automatically get deleted by
+  `with-model-cleanup`. Conditions should be a HoneySQL `:where` clause."
+  {:arglists '([model])}
+  type)
+
+(defmethod with-model-cleanup-additional-conditions :default
+  [_]
+  nil)
+
+(defmethod with-model-cleanup-additional-conditions (type Collection)
+  [_]
+  ;; NEVER delete personal collections for the test users.
+  [:or
+   [:= :personal_owner_id nil]
+   [:not-in :personal_owner_id (set (map (requiring-resolve 'metabase.test.data.users/user->id)
+                                         @(requiring-resolve 'metabase.test.data.users/usernames)))]])
+
 (defn do-with-model-cleanup [models f]
   {:pre [(sequential? models) (every? t.models/model? models)]}
+  (test-runner/assert-test-is-not-parallel "with-model-cleanup")
```

Comment:
Extra tooling to make sure we don't delete test users personal collections when we use `with-model-cleanup`